closed/custom/Docs.gmk

```diff
@@ -116,3 +116,174 @@ $(OPENJ9_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER)
+
+OPENJDK_MODULE_REFERENCE_TEXT := \
+  These are the OpenJ9 extensions to this module. Refer to <a href="$(JAVASE_BASE_URL)">$(JAVASE_BASE_URL)</a> for the reference implementation documentation.
+
+MODULE_DESCRIPTION_FIRST_LINE  := <section role="region">
+MODULE_DESCRIPTION_SECOND_LINE := MODULE DESCRIPTION
+MODULE_DESCRIPTION_MARKER_TEXT := javadoc_marker_text
+MODULE_DESCRIPTION_LAST_LINE   := </section>
+MODULE_DESCRIPTION_REPLACE_TEXT := \
+  <section class="module-description" id="module.description"> \
+    <div class="block"> \
+      <dl class="notes"> \
+        <dd>$(OPENJDK_MODULE_REFERENCE_TEXT)</dd> \
+      </dl> \
+    </div> \
+  </section>
+
+OPENJDK_JDK_MANAGEMENT_SUMMARY := Defines JDK-specific management interfaces for the JVM.
+OPENJDK_JDK_MANAGEMENT_SUMMARY_ERROR_TEXT := \
+Error: cannot find text '$(OPENJDK_JDK_MANAGEMENT_SUMMARY)' to replace in generated javadoc file $(OPENJ9_ONLY_INDEX_HTML). \
+Check the openjdk jdk.management/module.info.java file and adjust closed/custom/Docs.gmk as required.
+
+# sed_replace_block Replaces a multiline block with replacement text.
+# sed matching  is greedy.
+# $1 should be unique (otherwise all lines from the first occurrence of $1 are deleted).
+sed_replace_block = $(SED) '/$(subst /,\/,$(1))/,/$(subst /,\/,$(2))/{/$(subst /,\/,$(2))/ s/.*/$(subst /,\/,$(3))/; t; d}'
+
+# sed_replace_regex replaces a regex found anywhere in the input with the replacement
+# text (use to replace matches over multiple lines).
+# $1 is the regex to search for, $2 is the replacement text.
+sed_replace_regex = $(SED) -n '1h;1!H;$${g;s/$(subst /,\/,$(1))/$(subst /,\/,$(2))/;p}'
+
+# sed_replace_first is a simple string replace, first occurrence only
+sed_replace_first = $(SED) 's/$(subst /,\/,$(1))/$(subst /,\/,$(2))/'
+
+ALL_PACKAGE_INFO_FILELIST := $(call FindFiles, $(TOP_SRC_DIRS),package-info.java)
+ALL_PACKAGE_INFO_FILELIST += $(call FindFiles, $(TOP_SRC_DIRS),package.html)
+OPENJ9_ONLY_PACKAGE_INFO_FILELIST  := $(filter %package-info.java, $(OPENJ9_ONLY_FILELIST))
+OPENJDK_ONLY_PACKAGE_INFO_FILELIST := $(filter-out $(OPENJ9_ONLY_PACKAGE_INFO_FILELIST),$(ALL_PACKAGE_INFO_FILELIST))
+OPENJDK_ONLY_PKGLIST := $(subst /package.html,,$(subst /package-info.java,,$(subst /share/classes,,$(subst $(TOPDIR)/src/,,$(OPENJDK_ONLY_PACKAGE_INFO_FILELIST)))))
+
+OPENJ9_STUBS_DIR := $(SUPPORT_OUTPUTDIR)/openj9-docs/_javadoc_openj9_stub_files
+OPENJ9_ONLY_PACKAGE_STUB_DESC := \
+  /** These are the classes within this package modified by OpenJ9. Refer to <a href="$(JAVASE_BASE_URL)">$(JAVASE_BASE_URL)</a> for the reference implementation documentation.*/
+
+$(foreach dir,$(OPENJDK_ONLY_PKGLIST), \
+ $(shell $(MKDIR) -p $(OPENJ9_STUBS_DIR)/$(dir); \
+         $(ECHO) "$(OPENJ9_ONLY_PACKAGE_STUB_DESC)" >$(OPENJ9_STUBS_DIR)/$(dir)/package-info.java; \
+         $(ECHO) "package $(subst /,.,$(subst $(firstword $(subst /, ,$(dir)))/,,$(dir)));" >>$(OPENJ9_STUBS_DIR)/$(dir)/package-info.java) \
+)
```

Comment:
This work should occur in an rule rather than as a side-effect of reading the makefile.

Reply:
Reworked and moved inside the build target.